### PR TITLE
Clarify VIP vs DNSRR options

### LINKED ISSUE #4713: DNSRR Section seems wrong
File: [engine/swarm/networking.md](https://docs.docker.com/engine/swarm/networking/), CC @mstanleyjones

> DNS round-robin is useful in cases where you want to use your own load balancer. 

It seems it should be:
DNS round-robin is useful in cases where you don't want to use your own load balancer.

### PR DESCRIPTION
**Pull request by gbarr01**
Clarify the two different methods of service discovery (in paragraph above the bullets).

Fixes #4713

### CODE REVIEW

File: engine/swarm/networking.md [hunk 1]
```diff
@@ -233,13 +233,16 @@ from any swarm node which is joined to the swarm and is in a `running` state.
...
-DNS round robin (DNSRR). You can configure this per service.
+services on the same network. For instance, if you have a [WordPress service
+that stores its data in a MySQL
+service](http://training.play-with-docker.com/swarm-service-discovery/), and
+they are connected to the same overlay network, you do not need to publish the
+MySQL port to the client, only the WordPress HTTP port.
+
+Service discovery can work in two different ways: internal connection-based
+load-balancing at Layers 3 and 4 using the embedded DNS and a virtual IP (VIP),
+or external and customized request-based load-balacning at Layer 7 using DNS
```
Comment (thaJeztah) [inline engine/swarm/networking.md]: typo "balacning"

Comment (gbarr01) [reply engine/swarm/networking.md]: Thank you, @thaJeztah !

File: engine/swarm/networking.md [hunk 2]
```diff
@@ -233,13 +233,16 @@ from any swarm node which is joined to the swarm and is in a `running` state.
...
 needing to know how many nodes are participating in the service or their
 IP addresses or ports. You don't need to publish ports which are used between
-services on the same network. For instance, if you have a WordPress service
-which stores its data in a MySQL service, and they are connected to the same
-overlay network, you do not need to publish the MySQL port to the client, only
-the WordPress HTTP port.
-
-Service discovery can work in two different ways, using a virtual IP (VIP) or
-DNS round robin (DNSRR). You can configure this per service.
+services on the same network. For instance, if you have a [WordPress service
```
Comment (mstanleyjones) [inline engine/swarm/networking.md]: This is personal preference, but I'd wrap after `if you have a` and then put the whole link text and link all on one line, so your next newline would be after the `,`. It breaks wrapping  conventions but it makes you less likely to accidentally break your link by having an inadvertent space or something at the wrong place.

**Comment by gbarr01** [comment]
Hi, @thaJeztah , I fixed the typo and am going to merge. Please let me know if you disagree or see other problems. Cheers.

**Pull request merged**